### PR TITLE
all homework week 20

### PR DESCRIPTION
**Pull request by daleel1**

### CODE REVIEW

File: src/components/Results.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import moment from "moment";
+
+const Results = (props) => {
+  const res = props.results;
```
Comment (Mullefa) [inline src/components/Results.js]: Not sure if this intermediate value is needed? Could just do `props.results.map()`

File: src/components/Results.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          <th>title</th>
+          <th>First Name</th>
+          <th>Surname </th>
+          <th>Email </th>
+          <th>Room ID </th>
+          <th>Check in Date </th>
+          <th>Check out Date </th>
+          <th>number of days</th>
+        </tr>
+        {res.map(function (result){
```
Comment (Mullefa) [inline src/components/Results.js]: Think an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) is more idiomatic.

File: src/components/Results.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        return (<tr>
+          <td>{result.title}</td>
+          <td>{result.firstName}</td>
+          <td>{result.surname}</td>
+          <td>{result.email}</td>
+          <td>{result.roomId}</td>
+          <td>{result.checkInDate}</td>
+          <td>{result.checkOutDate}</td>
+          <td>{moment(result.checkOutDate).diff(moment(result.checkInDate), 'days')}</td>
```
Comment (Mullefa) [inline src/components/Results.js]: Nice! Maybe cleaner to factor it out as a utility function though e.g. `dayDiff()`

File: src/components/Search.js [hunk 4]
```diff
@@ -11,9 +12,18 @@ const Search = props => (
           <label htmlFor="customerId">Customer Id</label>
           <div className="search-row">
             <input type="text" id="customerId" className="form-control" placeholder="Customer Id" />
-            <button className="btn btn-primary fn-submit-name">Search IDs</button>
+            <Button name="Search IDs"/>
```
Comment (Mullefa) [inline src/components/Search.js]: `buttonText` might be a clearer / more accurate prop key name than `name`

File: src/containers/Bookings.js [hunk 5]
```diff
@@ -1,21 +1,24 @@
 import React, { Component } from 'react';
 import Search from '../components/Search.js';
 // import Results from '../components/Results.js';
-// import FakeBookings from '../data/fakeBookings.json';
+import Results from '../components/Results.js';
 
+// import FakeBookings from '../data/fakeBookings.json';
```
Comment (Mullefa) [inline src/containers/Bookings.js]: Comment (`// ...`) can be removed

File: src/components/Results.js (hunk 2 above)
Comment (Mullefa) [reply src/components/Results.js]: A better function argument name could be `reservation` or `booking` (rather than `result`)

File: package.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "moment": "^2.22.2",
```
Comment (Mullefa) [inline package.json]: Nice!

Comment (reber12) [reply package.json]: @Mullefa thanks for your comments on my PR, and for taking the time to review my work

**Pull request closed without merging**